x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/slm/SnapshotRetentionTaskTests.java

```diff
@@ -166,6 +174,73 @@ public void testRetentionTask() throws Exception {
+             ClusterService clusterService = ClusterServiceUtils.createClusterService(threadPool);
+             Client noOpClient = new NoOpClient("slm-test")) {
+
+            SnapshotLifecyclePolicy policy = new SnapshotLifecyclePolicy("policy", "snap", "1 * * * * ?",
+                "repo", null, new SnapshotRetentionConfiguration(null, null,1));
+
+            ClusterState state = createState(policy);
+            state = ClusterState.builder(state)
+                .metaData(MetaData.builder(state.metaData())
+                    .transientSettings(Settings.builder()
+                        .put(LifecycleSettings.SLM_RETENTION_DURATION, "500ms")
+                        .build())).build();
+            ClusterServiceUtils.setState(clusterService, state);
+
+            final SnapshotInfo snap1 = new SnapshotInfo(new SnapshotId("name1", "uuid1"), Collections.singletonList("index"),
+                0L, null, 1L, 1, Collections.emptyList(), true, Collections.singletonMap("policy", "policy"));
+            final SnapshotInfo snap2 = new SnapshotInfo(new SnapshotId("name2", "uuid2"), Collections.singletonList("index"),
+                1L, null, 2L, 1, Collections.emptyList(), true, Collections.singletonMap("policy", "policy"));
+            final SnapshotInfo snap3 = new SnapshotInfo(new SnapshotId("name3", "uuid3"), Collections.singletonList("index"),
+                2L, null, 3L, 1, Collections.emptyList(), true, Collections.singletonMap("policy", "policy"));
+            final SnapshotInfo snap4 = new SnapshotInfo(new SnapshotId("name4", "uuid4"), Collections.singletonList("index"),
+                3L, null, 4L, 1, Collections.emptyList(), true, Collections.singletonMap("policy", "policy"));
+            final SnapshotInfo snap5 = new SnapshotInfo(new SnapshotId("name5", "uuid5"), Collections.singletonList("index"),
+                4L, null, 5L, 1, Collections.emptyList(), true, Collections.singletonMap("policy", "policy"));
+
+            final Set<SnapshotId> deleted = ConcurrentHashMap.newKeySet();
+            // We're expected two deletions before they hit the "taken too long" test, so have a latch of 2
+            CountDownLatch latch = new CountDownLatch(2);
+            OverrideDeleteSnapshotRetentionTask retentionTask = new OverrideDeleteSnapshotRetentionTask(noOpClient, clusterService,
+                () -> {
+                    List<SnapshotInfo> snaps = Arrays.asList(snap1, snap2, snap3, snap4, snap5);
+                    logger.info("--> retrieving snapshots [{}]", snaps);
+                    return Collections.singletonMap("repo", snaps);
+                },
+                (repo, snapshotId) -> {
+                    // Don't pause until snapshot 2
+                    if (snapshotId.equals(snap2.snapshotId())) {
+                        try {
+                            logger.info("--> pausing for 501ms while deleting snap2 to simulate deletion past a threshold");
+                            Thread.sleep(501);
```

Comment:
I'm okay with this going in as-is for now, but I think we should switch this to use a now provider that we can manipulate for the test, rather than `Thread.sleep` before merging to master.